src/main/java/net/neoforged/neoforge/common/extensions/ILevelReaderExtension.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.common.extensions;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Registry;
+import net.minecraft.core.RegistryAccess;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.world.level.LevelReader;
+
+public interface ILevelReaderExtension extends RegistryAccess {
```

Comment:
Not a fan of making it extend `RegistryAccess` directly, as that brings weird methods like `freeze()` in scope.

Reply:
Previously I had the methods duplicated. I believe it could be narrowed in scope to only extend HolderLookup.Provider, which afaik is sufficient for the majority of usecases.

Reply:
A bit of duplication is fine. (And generally preferable to inheritance spaghetti, IMO).

Reply:
While im not sure how useful or relevant it might be, it would also let you pass `level` in place of methods that require a Provider. Since levels are typically the only way to obtain a registry in many scopes to begin with, I don't think its *too* smelly to allow a level to be treated as a provider directly, but I won't die on the hill, and those use cases are far fewer anyway. If duplicated methods are better for maintainability im happy with that.

Reply:
I don't think there is much value when one could instead just pass `level.registries()`.

Reply:
Sounds good. I removed the extension and added the methods to `IlevelReaderExtension` which calls those same methods from the HolderExtension.